nexus/test-utils/src/lib.rs

```diff
@@ -870,41 +922,51 @@ impl<'a, N: NexusServer> ControlPlaneTestContextBuilder<'a, N> {
                             );
                         }
                     }
-                    // Populate extra fake disks, giving each sled 10 total.
-                    if disks.len() < 10 {
-                        for _ in disks.len()..10 {
-                            disks.push(BlueprintPhysicalDiskConfig {
-                                disposition:
-                                    BlueprintPhysicalDiskDisposition::InService,
-                                identity: omicron_common::disk::DiskIdentity {
-                                    vendor: "nexus-tests".to_string(),
-                                    model: "nexus-test-model".to_string(),
-                                    serial: format!(
-                                        "nexus-test-disk-{disk_index}"
-                                    ),
-                                },
-                                id: PhysicalDiskUuid::new_v4(),
-                                pool_id: ZpoolUuid::new_v4(),
-                            });
-                            disk_index += 1;
-                        }
-                    }
-                    blueprint_disks.insert(
-                        sled_id,
-                        BlueprintPhysicalDisksConfig {
-                            generation: Generation::new().next(),
-                            disks,
-                        },
-                    );
-                    blueprint_datasets.insert(
+                } else {
+                    blueprint_zones.insert(
                         sled_id,
-                        BlueprintDatasetsConfig {
+                        BlueprintZonesConfig {
                             generation: Generation::new().next(),
-                            datasets,
+                            zones: IdMap::new(),
                         },
                     );
                 }
+
+                // Populate extra fake disks, giving each sled 10 total.
+                if disks.len() < 10 {
+                    for _ in disks.len()..10 {
+                        disks.push(BlueprintPhysicalDiskConfig {
+                            disposition:
+                                BlueprintPhysicalDiskDisposition::InService,
+                            identity: omicron_common::disk::DiskIdentity {
+                                vendor: "nexus-tests".to_string(),
+                                model: "nexus-test-model".to_string(),
+                                serial: format!("nexus-test-disk-{disk_index}"),
+                            },
+                            id: PhysicalDiskUuid::new_v4(),
+                            pool_id: ZpoolUuid::new_v4(),
+                        });
+                        disk_index += 1;
+                    }
+                }
+
+                blueprint_disks.insert(
```

Comment:
Just an FYI, this just changed to use IdMaps in https://github.com/oxidecomputer/omicron/pull/7352, so you'll need a bit more cleanup here. Sorry!

Reply:
No problem :)